### PR TITLE
Implements correct semantics in the test setup for members integration tests

### LINKED ISSUE #680: [rfc] Implement correct semantics in the test setup for members integration tests
Ideally in the test setup part
1. beforeEach should populate data(if any) - **Setup**
2. afterEach should perform the cleanup tasks(if any) - **Teardown**

But currently, the test setup in member integration tests is not following this. In fact the opposite of this is being done here.

One reason for doing so was that, in a describe block the first test required DB to be clean and the following tests required 1 or more users to be present in the DB. 
This is bad since a test should not depend on the sequence it's running. Because of this ,in the future, if we want to run the tests in parallel we won't be able to do so.

### PR DESCRIPTION
**Pull request by samarpan1738**
Rearranged beforeEach and afterEach blocks

Closes #680

### CODE REVIEW

File: test/integration/members.test.js [hunk 1]
```diff
@@ -38,41 +38,35 @@ describe("Members", function () {
   let jwt;
 
   afterEach(async function () {
-    await addUser();
-  });
-
```
Comment (rohan09-raj) [inline test/integration/members.test.js:43]: Thanks for fixing this

File: test/integration/members.test.js [hunk 2]
```diff
@@ -38,41 +38,35 @@ describe("Members", function () {
...
 
     it("Should return empty array if no user is found", function (done) {
-      chai
-        .request(app)
-        .get("/members")
-        .end((err, res) => {
-          if (err) {
-            return done(err);
-          }
+      cleanDb().then(() => {
```
Comment (rohan09-raj) [inline test/integration/members.test.js:53]: Is after each doesn't taking care of this ??

File: test/integration/members.test.js [hunk 3]
```diff
@@ -135,25 +129,28 @@ describe("Members", function () {
...
+
     it("Should return empty array if no idle member is found", function (done) {
-      chai
-        .request(app)
-        .get("/members/idle")
-        .end((err, res) => {
-          if (err) {
-            return done(err);
-          }
+      cleanDb().then(() => {
```
Comment (rohan09-raj) [inline test/integration/members.test.js:137]: Is after each doesn't taking care of this ??

**Review by rohan09-raj — CHANGES_REQUESTED** [review]

File: test/integration/members.test.js (hunk 3 above)
Comment (samarpan1738) [reply test/integration/members.test.js:137]: After each cleans the DB but then beforeEach adds the users in the DB.

File: test/integration/members.test.js (hunk 2 above)
Comment (samarpan1738) [reply test/integration/members.test.js:53]: After each cleans the DB but then beforeEach adds the users in the DB.

Comment (rohan09-raj) [reply test/integration/members.test.js:53]: Continuing the conversation from here as its the same comment https://github.com/Real-Dev-Squad/website-backend/pull/681#discussion_r945517685

File: test/integration/members.test.js (hunk 3 above)
Comment (rohan09-raj) [reply test/integration/members.test.js:137]: I'm saying you have already added cleanDb in afterEach, so calling this function here and doing something on resolving the promise, is this necessary to do so ??

Comment (samarpan1738) [reply test/integration/members.test.js:137]: As discussed we'll we creating separate describe blocks for the 2 cases : when there are no users in the DB and when there are users in DB

File: test/integration/members.test.js [hunk 4]
```diff
@@ -156,31 +151,32 @@ describe("Members", function () {
...
-    it("Get all the idle members in the database", function (done) {
-      chai
-        .request(app)
-        .get("/members/idle")
-        .end((err, res) => {
-          if (err) {
-            return done(err);
-          }
+    it("Should return all the idle members", function (done) {
+      addUser().then(() => {
```
Comment (rohan09-raj) [inline test/integration/members.test.js:155]: Can we set this in separate `describe` in a `before`  as we discussed earlier for previous comments ?

Comment (samarpan1738) [reply test/integration/members.test.js:155]: Chai recommends that if a describe has only 1 it block better don't use describe for it. It gives lint error also

Comment (samarpan1738) [reply test/integration/members.test.js:155]: Updated the above comment as well
https://github.com/lo1tuma/eslint-plugin-mocha/blob/master/docs/rules/no-hooks-for-single-case.md

**Review by rohan09-raj — APPROVED** [review]

**Pull request merged**